src/redux/books/books.js

```diff
@@ -1,42 +1,119 @@
-import { nanoid } from '@reduxjs/toolkit';
-
-const initialState = [
-  {
-    id: '2',
-    title: 'My Book of Bible Stories',
-    author: 'jw.org',
-  },
-  { id: '657890', title: 'Your family can be happy', author: 'John Doe' },
-];
+export const ADD_BOOK = 'ADD_BOOK';
+export const GET_ALL_BOOKS = 'GET_ALL_BOOKS';
+export const REMOVE_BOOK = 'REMOVE_BOOK';
+const FETCH_BOOKS = 'FETCH_BOOKS';
+const FETCH_BOOKS_ERROR = 'FETCH_BOOKS_ERROR';
+const FETCH_BOOKS_LOADING = 'FETCH_BOOKS_LOADING';
+const API_KEY = 'FYlvcWEDlWEYN00KzCCr';
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net';
+export const URL = `${BASE_URL}/bookstoreApi/apps/${API_KEY}/books`;
 
 export const addBook = (book) => ({
-  type: 'ADD_BOOK',
-  book,
+  type: ADD_BOOK,
+  payload: book,
 });
 
 export const removeBook = (id) => ({
-  type: 'REMOVE_BOOK',
-  id,
+  type: REMOVE_BOOK,
+  payload: id,
+});
+
+export const fetchBooks = (books) => ({
+  type: FETCH_BOOKS,
+  payload: books,
+});
+
+export const fetchBooksError = (error) => ({
+  type: FETCH_BOOKS_ERROR,
+  payload: error,
 });
 
+export const fetchBooksLoading = () => ({
+  type: FETCH_BOOKS_LOADING,
+});
+
+const initialState = {
+  books: [],
+  loading: false,
+  error: null,
+};
+
+export const getBooks = () => (dispatch) => {
+  dispatch(fetchBooksLoading());
+  fetch(URL)
+    .then((response) => response.json())
+    .then((data) => {
+      const newData = [];
+      Object.keys(data).forEach((key) => {
+        if (key) {
+          newData.push({
+            ...data[key][0],
+            item_id: key,
+          });
+        }
+      });
+      dispatch(fetchBooks(newData));
+    })
+    .catch((error) => {
+      dispatch(fetchBooksError(error.message));
+    });
+};
+
+export const createBook = (book) => (dispatch) => {
+  fetch(URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  }).then(() => {
+    dispatch(addBook(book));
+  });
+};
+
+export const deleteBook = (id) => (dispatch) => {
+  fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({ item_id: id }),
+  }).then(() => {
+    dispatch(removeBook(id));
+  });
+};
+
```

Comment:
- [ ] Kindly as the requirements says make sure you use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.